src/index.ts

```diff
@@ -226,28 +262,34 @@ module.exports = (app: ChartProviderApp): Plugin => {
         if (!provider) {
           return res.sendStatus(404)
         }
-        switch (provider._fileFormat) {
-          case 'directory':
-            return serveTileFromFilesystem(
-              res,
-              provider,
-              parseInt(z),
-              parseInt(x),
-              parseInt(y)
-            )
-          case 'mbtiles':
-            return serveTileFromMbtiles(
-              res,
-              provider,
-              parseInt(z),
-              parseInt(x),
-              parseInt(y)
-            )
-          default:
-            console.log(
-              `Unknown chart provider fileformat ${provider._fileFormat}`
-            )
-            res.status(500).send()
+        if (provider.cached === true) {
+          return serveTileFromCache(res, provider, parseInt(z), parseInt(x), parseInt(y))
```

Comment:
nitpick, but I would parse x,y and z once instead of sprinkling the parseInts in all the three serve calls.